### ISSUE TITLE
Keras submodels are not supported 

### ISSUE THREAD

**Issue by katerina-merkulova** [intel/openfl#185]
As far as I know, problem occurs at serialization and deserialization steps. User should define get_config and from_config methods ([docs](https://keras.io/guides/serialization_and_saving/)). Then config requires an InputLayer, but submodel doesn't have it. It seems that the InputLayer is only created if user wraps model with functional API ([stackoverflow](https://stackoverflow.com/questions/58153888/how-to-set-the-input-of-a-keras-subclass-model-in-tensorflow)).

**Comment by alexey-gruzdev** [comment]
@katerina-merkulova could you please write down a simple example of such network definition which is not supported.

**Comment by katerina-merkulova** [comment]
Example of model:
~~~
class Model(tf.keras.Model):
    def __init__(self, **kwargs):
        super().__init__()

        self.lstm = LSTM(1000)
        self.d1 = Dense(1000, activation='tanh')
        self.d2 = Dense(10719, activation='softmax')

    def call(self, x):
        x = self.lstm
        x = self.d1(x)
        x = self.d2(x)
        return x
~~~
And these are my attempts to define required methods:
~~~
    def get_config(self):
        config = dict({'name': self.name})
        config['layers'] = [
            {'class_name': layer.__class__.__name__,
             'config': layer.get_config()}
            for layer in self._self_tracked_trackables
        ]
        return config
    
    @classmethod
    def from_config(cls, config):
        return cls(**config)
~~~
